server/api/db/sqldb/db.py

```diff
@@ -4180,6 +4182,179 @@ def create_data_version(self, session, version):
+
+        self._store_notifications(
+            session, AlertConfig, alert.notifications, alert_record.id, alert.project
+        )
+
+        state = AlertConfig.AlertState(count=0, parent_id=alert_record.id)
+        self._upsert(session, [state])
+        return self._transform_alert_config_record_to_schema(alert_record)
+
+    def store_alert(
+        self, session, alert: mlrun.common.schemas.AlertConfig
+    ) -> mlrun.common.schemas.AlertConfig:
+        alert_record = self._get_alert_by_id(session, alert.id)
+        alert_record.full_object = alert.dict()
+        alert_state = self.get_alert_state(session, alert.id)
+
+        self._delete_alert_notifications(session, alert.name, alert, alert.project)
+        self._store_notifications(
+            session, AlertConfig, alert.notifications, alert_record.id, alert.project
+        )
+
+        self._upsert(session, [alert_record, alert_state])
+        return self._transform_alert_config_record_to_schema(
+            self._get_alert_by_id(session, alert_record.id)
+        )
+
+    def delete_alert(self, session, project: str, name: str):
+        self._delete(session, AlertConfig, project=project, name=name)
+
+    def list_alerts(
+        self, session, project: str = None
+    ) -> list[mlrun.common.schemas.AlertConfig]:
+        query = self._query(session, AlertConfig)
+
+        if project and project != "*":
+            query = query.filter(AlertConfig.project == project)
+
+        alerts = list(map(self._transform_alert_config_record_to_schema, query.all()))
+        for alert in alerts:
+            self.enrich_alert(session, alert)
+        return alerts
+
+    def get_alert(
+        self, session, project: str, name: str
+    ) -> mlrun.common.schemas.AlertConfig:
+        return self._transform_alert_config_record_to_schema(
+            self._get_alert(session, name, project)
+        )
+
+    def get_alert_by_id(
+        self, session, alert_id: int
+    ) -> mlrun.common.schemas.AlertConfig:
+        resp = self._query(session, AlertConfig, id=alert_id).one_or_none()
+        return self._transform_alert_config_record_to_schema(resp)
+
+    def enrich_alert(self, session, alert: AlertConfig):
+        state = self.get_alert_state(session, alert.id)
+        alert.state = (
+            mlrun.common.schemas.AlertActiveState.ACTIVE
+            if state.active
+            else mlrun.common.schemas.AlertActiveState.INACTIVE
+        )
+        alert.count = state.count
+
+        def _adjust_obj(notif):
+            notif = notif.to_dict()
+            del notif["secret_params"]
+
+            if not isinstance(notif["when"], list):
+                notif["when"] = [notif["when"]]
+            return notif
+
+        alert.notifications = [
+            mlrun.common.schemas.notification.Notification(**_adjust_obj(x))
+            for x in self._get_db_notifications(
+                session, AlertConfig, parent_id=alert.id
+            )
+        ]
```

Comment:
Please don't use such generic/shortened variable and method names and please add an explanation, why remove the secret params?
```suggestion
        def _enrich_notification(_notification):
            _notification = _notification.to_dict()
            del _notification["secret_params"]

            if not isinstance(_notification["when"], list):
                _notification["when"] = [_notification["when"]]
            return _notification

        alert.notifications = [
            mlrun.common.schemas.notification.Notification(**_enrich_notification(notification))
            for notification in self._get_db_notifications(
                session, AlertConfig, parent_id=alert.id
            )
        ]
```